### DISCUSSION TITLE
Problem compiling libmetal for baremetal

### DISCUSSION THREAD

**Discussion by jmpapke** [OpenAMP/libmetal#241] [Q&A]
Hello,

I need to make a baremetal application which runs on a zynq7000 (mercury PE1 200 card) using yocto. I could compile everything for the linux side. 
Regarding the baremetal application running on the remote cpu I have chosen to use the linaro toolchain to compile the libmetal library. This is achieve using a zynq7.cmake file:
set (CMAKE_SYSTEM_PROCESSOR "arm" CACHE STRING "")
set (MACHINE                "zynq7"          CACHE STRING "")
set (CROSS_PREFIX           "${cross_compiler_path}/arm-eabi-" CACHE STRING "")
set (CMAKE_C_FLAGS          "-mcpu=cortex-a9 -mfpu=vfpv3 -mfloat-abi=hard" CACHE STRING "")
set(WITH_EXT_INCLUDES_FIND "off")
include (cross-generic-gcc)
set (CMAKE_SYSTEM_NAME "Generic"             CACHE STRING "")

However I still face a problem regarding this compilation:
  tmp-glibc/work/cortexa9t2hf-neon-leca-linux-gnueabi/libmetal-baremetal/1.0-r0/build/lib/include/metal/system/generic/./zynq7/sys.h:17:10: fatal error: xscugic.h: No such file or directory

Could somebody help me ?
Thanks in advance

**Comment by tnmysh** [comment]
Hello,

Thanks for your email.


On 7/6/23 11:44 AM, jmpapke wrote:
>
> Hello,
>
> I need to make a baremetal application which runs on a zynq7000 
> (mercury PE1 200 card) using yocto. I could compile everything for the 
> linux side.
> Regarding the baremetal application running on the remote cpu I have 
> chosen to use the linaro toolchain to compile the libmetal library. 
> This is achieve using a zynq7.cmake file:
> set (CMAKE_SYSTEM_PROCESSOR "arm" CACHE STRING "")
> set (MACHINE "zynq7" CACHE STRING "")
> set (CROSS_PREFIX "${cross_compiler_path}/arm-eabi-" CACHE STRING "")
> set (CMAKE_C_FLAGS "-mcpu=cortex-a9 -mfpu=vfpv3 -mfloat-abi=hard" 
> CACHE STRING "")
> set(WITH_EXT_INCLUDES_FIND "off")
> include (cross-generic-gcc)
> set (CMAKE_SYSTEM_NAME "Generic" CACHE STRING "")
>
> However I still face a problem regarding this compilation:
> tmp-glibc/work/cortexa9t2hf-neon-leca-linux-gnueabi/libmetal-baremetal/1.0-r0/build/lib/include/metal/system/generic/./zynq7/sys.h:17:10: 
> fatal error: xscugic.h: No such file or directory
>
Unfortunately as of now you have to use Xilinx toolchain to build 
baremetal firmware.

xscugic.h is from Xilinx's proprietary library that has baremetal BSP. 
It is Something that cannot be compiled with

Linaro toolchain due to dependencies of Xilinx toolchain.


Thanks.

> Could somebody help me ?
> Thanks in advance
>
> —
